### ISSUE TITLE
False positive: useless-parent-delegation on '__init__' method of class derived from 'Exception'

### ISSUE THREAD

**Issue by dnpetrov** [pylint-dev/pylint#9994]
### Bug description

```python
useless-parent-delegation is reported on '__init__' method of class derived from 'Exception'.
This '__init__' method, though, accepts no arguments except for 'self'.
For regular classes following the same pattern (base class '__init__' accepts arbitrary additional arguments, derived class '__init__' accepts only 'self'), no warning is reported.

a.py:

class Base:
    def __init__(self, *args, **kwargs):
        pass


class Derived(Base):
    # No warning
    def __init__(self):
        super().__init__()


class MyException(Exception):
    # W0246: Useless parent or super() delegation in method '__init__' (useless-parent-delegation)
    def __init__(self):
        super().__init__()

```
```


### Configuration

_No response_

### Command used

```shell
pylint a.py
```

### Pylint output

```python
************* Module a
a.py:14:4: W0246: Useless parent or super() delegation in method '__init__' (useless-parent-delegation)
```

### Expected behavior

No warning for MyException '__init__'.

### Pylint version

```shell
pylint 3.2.5
astroid 3.2.3
Python 3.10.12 (main, Jul 29 2024, 16:56:48) [GCC 11.4.0]
```

### OS / Environment

Ubuntu 22.04.4
